app/src/main/java/fr/free/nrw/commons/upload/UploadActivity.java

```diff
@@ -143,6 +146,7 @@ public class UploadActivity extends BaseActivity implements UploadContract.View,
      */
     public static HashMap<Place,Boolean> nearbyPopupAnswers;
 
+    public boolean hasAllPermissions = true;
```

Comment:
Please add javadoc explaining what it means exactly when true or when false.

Reply:
Better avoid public members. Would you mind making it private and creating a setter and getter?

Reply:
Sure, I have added the suggested changes in newer commits.